[26기 옥원철] [Mission 5] ID, PW Validation 기능 추가 / [Mission 6] 댓글 삭제 기능 추가 / 셀프리팩토링

okcleff opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약

  - [Mission 5, 6] 및 셀프 리팩토링 수행
		 
## 수정 사항들 자세한 내용

 1. [Mission 5] ID, PW Validation 기능 추가
   - ID에 @ 포함, PW 5자리 이상 입력 시 로그인 버튼 활성화

 2. [Mission 6] 댓글 삭제 기능 추가
   - x 버튼 입력 시 댓글 삭제

 3. 셀프 리팩토링 수행
   - id, class명 camelCase로 통일
   - img tag에 alt 속성 부여
   - CSS 속성을 권장되는 순서로 재정렬

## 기타 질문 및 특이 사항

- 지속적으로 셀프 리팩토링 할 예정입니다.
- [Mission 7, 8, 9] 추가 기능 구현 예정입니다.

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

joonsikyang commented:
라벨 달아주세요!

Jun4928 commented on students/13th/seosuyeon/js/main.js:
```diff
@@ -3,22 +3,52 @@ const commentInput = document.querySelector('#comment');
+    createBtnWrap.className = 'comments_btnwrap';
+    createLikeBtn.className = 'comments_likebtn';
+    createDeleteBtn.className = 'comments_deletebtn';
+    createBtnI.className = 'far fa-heart';
```
innerHTML 속성을 사용해서 리팩토링 해 보시겠어요?
일일이 모든 태그를 다 생성할 필요 없습니다.

Jun4928 commented on students/13th/seosuyeon/js/main.js:
```diff
@@ -3,22 +3,52 @@ const commentInput = document.querySelector('#comment');
+
+    createBtnWrap.appendChild(createLikeBtn);
+    createLikeBtn.appendChild(createBtnI);
+    createBtnWrap.appendChild(createDeleteBtn);
```
여기도 innerHTML 속성과 template literal 을 사용해서 좀 더 깔끔하게 코드를 구현할 수 있겠네요! 🙌

Jun4928 commented on students/13th/seosuyeon/js/main.js:
```diff
@@ -3,22 +3,52 @@ const commentInput = document.querySelector('#comment');
 // Events
 commentBtn.addEventListener('click', addComment);
+commentLikeBtn.addEventListener('click', changeColor);
+
```
이렇게 전역변수로 빼서 관리하는 것 👍

Jun4928 commented on students/13th/seosuyeon/main.html:
```diff
@@ -67,10 +68,29 @@
                     <ul id="comments_list" class="comments_list">
-                      <li class="comments_comment"><a class="comments_userlink" href="#">wecode</a>위워크에서 진행한 베이킹 클래스</li>
-                      <li class="comments_comment"><a class="comments_userlink" href="#">wecode</a>두번째</li>
+
```
태그간에 이런 불필요한 띄어쓰기는 삭제 해 주세요 🔥

Jun4928 commented on students/13th/seosuyeon/js/login.js:
```diff
@@ -12,5 +12,5 @@ function changeLoginBtnColor() {
     let inputPwValue = document.getElementById('pw').value;
 
-    (inputIdValue.length && inputPwValue.length > 5) ? loginBtn.style.backgroundColor = '#0095f6' : loginBtn.style.backgroundColor = '#b2dffc'
+    (inputIdValue.includes('@') && inputPwValue.length > 5) ? loginBtn.style.backgroundColor = '#0095f6' : loginBtn.style.backgroundColor = '#b2dffc'
```
삼항연산자 사용 잘 하셨네요! 
하지만 중복되는 코드(변수에 값을 대입하는 코드)가 있는데 다음과 같이 리팩토링 할 수 있습니다

**Hint**
```js
loginBtn.style.backgroundColor = (inputIdValue.includes('@') && inputPwValue.length > 5) ?  '#0095f6' :  '#b2dffc'

```

suyeon-DV replied:
감사합니다. 
중복 되는 코드 줄여보고 싶었는데 어떻게 해야할지 잘 몰랐습니다. 
코드 보시느라 수고하셨습니다.

Jun4928 commented:
질문주신 사항에 대해서 답변 드립니다.

1. main.css 에 관한 사항
이해하고 계신대로, main.css 에서는 프로젝트 전체에 중복되는 사항들을 변수화 해서 관리하는 것이 맞습니다. 해당내용은 [sass](https://sass-lang.com/) 를 배우게 되면 더 다양하게 활용할 수 있으니 지금 단계에서는 넘어가도 좋습니다.

2. class 작명 에 관한 사항
클래스네임을 작명해야 하는 상황엥서는 태그네임을 사용하지 않습니다. 
[css 셀렉터](https://www.w3schools.com/cssref/css_selectors.asp)를 사용하면 태그와 클래스네임을 다음과 같이 한 번에 셀렉트 할 수 있습니다.
```css
.div.account {
  properties..
}
```

Jun4928 commented:
수연님 전반적으로 코드가 굉장히 좋네요! (변수명, 클래스네임, html 구조 등.. 고심한 흔적이 보입니다)
잘 하시고 계시는데, 스스로 다독여 주세요! 🙌👍

저번에 이야기 나눈대로 급한마음 보다
차근차근 생각하는 시간 가지면서 즐겁게 코딩했으면 좋겠습니다 😊

okcleff committed:
ADD :
 [Mission 1, 2] 로그인 페이지 및 유효성 검사 완료
   - 로그인 페이지 레이아웃 작성 완료
   - ID & PW 입력에 따른 유효성 검사 기능 완료

 [Mission 3] 메인 페이지 레이아웃 작성중
   - 메인 페이지 레이아웃 작성중

okcleff committed:
ADD :
[Mission 3, 4] 메인 페이지 레이아웃 및 댓글 기능 구현 완료
 - 메인 페이지 레이아웃 작성
 - 엔터키 입력 또는 '게시'버튼 클릭을 통한 댓글 추가 기능 구현

REMOVE :
 - 불필요한 이미지 파일 삭제

okcleff committed:
ADD : [Mission 5] ID, PW Validation 기능 추가, [Mission 6] 댓글 삭제 기능 추가
REFACTOR : 코드 셀프 리팩토링
 - id, class명 camelCase로 통일
 - img tag에 alt 속성 부여
 - CSS 속성을 권장되는 순서로 재정렬
